src/model/names.ml

```diff
@@ -31,12 +48,12 @@ module Name : Name = struct
 
   let to_string = function
     | Std s -> s
-    | Internal (s, i) -> Printf.sprintf "$%s$%d" s i
+    | Internal (s, i) -> Printf.sprintf "{%s{%d" s i
 
   let to_string_unsafe = function Std s -> s | Internal (s, _i) -> s
 
   let of_string s =
-    if String.length s > 0 && s.[0] = '$' then raise (Invalid_argument s)
+    if String.length s > 0 && s.[0] = '{' then raise (Invalid_argument s)
```

Comment:
This is checking that we don't do `of_string (to_string name)` ? Perhaps we can remove this check and rename the function `std_of_string` (there is `internal_of_ident`).

Reply:
>  Perhaps we can remove this check

Hmm! I still find this right as is, could you please clarify.

Reply:
I mean that this function must not be called from the result of `to_string` so instead of checking that, we could rename the function to show that it has nothing to do with `to_string`. I suggested using the same convention as `internal_of_string` and `internal_of_ident` but `make_internal` and `make_std` would be better.

Reply:
I don’t mean to be adamant, but I think `of_string` and `of_ident` agree to the convention that I’ve seen being used in OCaml modules, for example `of_string` taking string as argument and returns t (for example).

> I mean that this function must not be called from the result of to_string

It’s also called from the result of `Ident.name x`.   `let of_ident id = of_string (Ident.name id)`

Reply:
`of_string` is supposed to make a `t` from the result of `to_string`. Here that's not what we want, we want a "constructor" kind of function, create a `t` from an input that mean something specific.

Reply:
Ah, I see!

Reply:
I think this should be addressed in a separate PR since it’s irrelevant to the PR and issue title

Reply:
Now that you made the change to the name of the function, I think this check is no longer needed.

Reply:
I'm a little nervous of removing the check - the original aim was to spot bugs, and the sort of bug it was looking for is certainly still possible. However, thinking a little more, I wonder whether we could incorporate this with a more general check - we could ensure at construction time that the names conform to OCaml's lexical conventions. This would include adding parentheses when necessary. This could catch bad references, such as `{!value-Foo}`.

Reply:
This function is no longer called `of_string` so at least the bug it was looking for is now less probable. I don't like adding such checks because it is checking for something that changes (OCaml's rules) and won't be correctly tested (because it's too complicated). I think this part of the code (and everything that is not the parser) shouldn't care for which characters are in these strings.

Reply:
To be clear - if we _do_ go along this path though, it should be in a different PR.

Reply:
Noted!